_posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md

```diff
@@ -0,0 +1,201 @@
+
+### Explore available entrypoints.
+
+Users can list all available entrypoints in a repo using the ```torch.hub.list()``` API.
+
+```python
+>>> torch.hub.list('pytorch/vision')
+>>>
+['alexnet',
+'deeplabv3_resnet101',
+'densenet121',
+...
+'vgg16',
+'vgg16_bn',
+'vgg19',
+ 'vgg19_bn']
+ ```
+
+Note that PyTorch Hub also allows auxillary entrypoints (other than pretrained models), e.g. ```bertTokenizer``` for preprocessing in the [BERT](https://pytorch.org/hub/huggingface_pytorch-pretrained-bert_bert/) models, to make the user workflow smoother.
+
+
+### Load a model
+
+Now that we know which models are available in the Hub, users can load a model entrypoint using the ```torch.hub.load()``` API. This only requires a single command without the need to install a wheel. In addition the ```torch.hub.help()``` API can provide useful information about how to instantiate the model.
+
+```python
+print(torch.hub.help('pytorch/vision', 'deeplabv3_resnet101'))
+model = torch.hub.load('pytorch/vision', 'deeplabv3_resnet101', pretrained=True)
+```
+
+It is also common that repo owners will want to continually add bug fixes or performance improvements. PyTorch Hub makes it super simple for users to get the latest update by calling:
+
+```python
+model = torch.hub.load(..., force_reload=True)
+```
+
+We believe this will help to alleviate the burden of repetitive package releases by repo owners and instead allow them to focus more on their research.
+It also ensures that, as a user, you are getting the freshest available models.
+
+On the contrary, stability is important for users. Hence, some model owners serve them from a specificed branch or tag, rather than the `master` branch, to ensure stability of the code.
+For example, `pytorch_GAN_zoo` serves them from the `hub` branch:
+
+```python
+model = torch.hub.load('facebookresearch/pytorch_GAN_zoo:hub', 'DCGAN', pretrained=True, useGPU=False)
+```
+
+
+### Explore a loaded model
+
+Once you have a model from PyTorch Hub loaded, you can use the following workflow to find out the available methods that are supported as well as understand better what arguments are requires to run it.
+
+
+```dir(model)``` to see all available methods of the model. Let's take a look at bertForMaskedLM's available methods.
+
+```python
+>>> dir(model)
+>>>
+['forward'
+...
+'to'
+'state_dict',
+]
+```
+
+```help(model.forward)``` provides a view into what arguments are required to make your loaded model run
+
+```python
+>>> help(model.forward)
+>>>
+Help on method forward in module pytorch_pretrained_bert.modeling:
+forward(input_ids, token_type_ids=None, attention_mask=None, masked_lm_labels=None)
+...
+```
+
+Note that the ```*args```, ```**kwargs``` passed to hub.load() are used to *instantiate* a model.
```

Comment:
Here we are mainly talking about "running" a model, so we probably want to make this more clear